R/methods-MChromatograms.R

```diff
@@ -635,3 +645,55 @@ setMethod("alignRt", signature = c(x = "MChromatograms", y = "Chromatogram"),
 #' @rdname MChromatograms-class
 setMethod("c", "MChromatograms",
           function(x, ...) .bind_rows_chromatograms(x, ...))
+
+#' @rdname MChromatograms-class
+setMethod("compareChromatograms",
+          signature = c(x = "MChromatograms", y = "missing"),
+          function(x, y, ALIGNFUN = alignRt, ALIGNFUNARGS = list(),
+                   FUN = cor, FUNARGS = list(use = "pairwise.complete.obs"),
+                   full = TRUE, ...) {
+              .compare_chromatograms(
+                  x, x, ALIGNFUN = alignRt, ALIGNFUNARGS = ALIGNFUNARGS,
+                  FUN = cor, FUNARGS = FUNARGS, full = full)
+          })
+
+#' @rdname MChromatograms-class
+setMethod("compareChromatograms",
+          signature = c(x = "MChromatograms", y = "MChromatograms"),
+          function(x, y, ALIGNFUN = alignRt, ALIGNFUNARGS = list(),
+                   FUN = cor, FUNARGS = list(use = "pairwise.complete.obs"),
+                   full = TRUE, ...) {
+              .compare_chromatograms(
+                  x, y, ALIGNFUN = alignRt, ALIGNFUNARGS = ALIGNFUNARGS,
+                  FUN = cor, FUNARGS = FUNARGS, full = full)
+          })
+
+.compare_chromatograms <- function(x, y, full = TRUE, ALIGNFUN = alignRt,
+                                   ALIGNFUNARGS = list(), FUN = cor,
+                                   FUNARGS = list(
+                                       use = "pairwise.complete.obs")) {
+    if (inherits(x, "MChromatograms")) {
+        if (ncol(x) > 1)
+            stop("Currently only single column MChromatograms are supported.")
+        x <- unlist(x)
+    }
+    if (inherits(y, "MChromatograms")) {
+        if (ncol(y) > 1)
+            stop("Currently only single column MChromatograms are supported.")
+        y <- unlist(y)
+    }
+    nx <- length(x)
+    ny <- length(y)
+
+    m <- matrix(NA_real_, nrow = nx, ncol = ny)
+    for (i in seq_len(nx)) {
+        for (j in seq_len(ny)) {
+            if (!full && i > j)
+                next
+            m[i, j] <- compareChromatograms(x[[i]], y[[j]], ALIGNFUN = ALIGNFUN,
```

Comment:
The lower triangle of the comparison matrix would be a mirror of the upper one, or? So it is a waste of time to calculate everything twice (if `full == TRUE`).
Why you don't use the same logic as in [`Spectra:::.compare_spectra_self`](https://github.com/rformassspectrometry/Spectra/blob/70fc21d0c35726d0c7e09e21f233dc5ab31fe7f2/R/Spectra-functions.R#L333-L358)

(Is `full` then even needed?)

Reply:
I would also guess the lower triangle to be the same than the upper triangle - so, I'll drop `full` and will calculate only the upper one (and the diagonal)

Reply:
In the updated version only the upper triangle is calculated and then copied to the lower